core/system/src/mpu/vmpu.c

```diff
@@ -216,11 +218,18 @@ int vmpu_fault_recovery_bus(uint32_t pc, uint32_t sp)
     if(!VMPU_FLASH_ADDR(pc))
        HALT_ERROR(NOT_ALLOWED, "This is not the PC (0x%08X) your were searching for", pc);
 
-    /* if the bus fault is imprecise we will seek back for ldrX/strX opcode */
-    if(SCB->CFSR & (1 << 10))
-        cnt_max = UVISOR_NOP_CNT;
-    else
+    /* check fault register; the following two configurations are allowed:
+     *   0x04 - imprecise data bus fault, no stacking/unstacking errors
+     *   0x82 - precise data bus fault, no stacking/unstacking errors */
+    if (SCB_BFSR == 0x82) {
```

Comment:
As both branches of the if use SCB_BFSR : it might be more readable to use a switch statement ?